### PR TITLE
Implement feature to rename items of the list

### PR DESCRIPTION
**Pull request by BohdanDudkaTD**
This PR adds logic to `rename` item of the list.
It's done by adding logic to check if the double click event occurs on the item. After it's fired - show `Input` where the user can type new name of the item. User can submit the name by pressing `Enter` or cancel the renaming process by pressing `Escape`. After a name is submitted - `RENAME_ITEM` action is dispatched. If it's successful - fetch a new list, otherwise - show error near the input field.

Result of the ESLint execution:
image

### CODE REVIEW

**Comment by BohdanDudkaTD** [comment]
@dkalinin, @AndriiSameliuk, PTAL.

File: app/view/page/file-list-page/index.js [hunk 1]
```diff
@@ -25,6 +25,12 @@ export class FileListPage extends ConfigurableComponent {
         this._setSelectedFile(undefined);
       }
     });
+    this.subscribe('stateChange.updatedFile', () => {
```
Comment (dkalinin) [inline app/view/page/file-list-page/index.js]: Let's not use updatedFile. Just check new file list if it has a file with id of selected file. If not – close file details. If yes — update file in info block.

File: app/view/component/file-list-item/index.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ export class FileListItem extends DragAndDropComponent {
    * @public
    */
   onDeleteButtonClick(callback) {
+    this._onDeleteButtoClickCallback = callback;
```
Comment (dkalinin) [inline app/view/component/file-list-item/index.js]: _onDeleteButto__n__ClickCallback

File: app/view/component/file-list-item/index.js [hunk 3]
```diff
@@ -163,18 +241,17 @@ export class FileListItem extends DragAndDropComponent {
...
-        event.stopPropagation();
-        this.dispatch(actions.GET_FOLDER_BY_ID, this._file.id)
-            .then(() => {
-              return this.dispatch(actions.FETCH_ITEMS);
-            })
-            .catch((error) => alert(error));
-      });
     }
 
+    this._listItem.addEventListener('click', (event) => {
```
Comment (dkalinin) [inline app/view/component/file-list-item/index.js]: Let's handle click on name only.

File: app/view/component/file-list-item/index.js [hunk 4]
```diff
@@ -163,18 +241,17 @@ export class FileListItem extends DragAndDropComponent {
...
+      event.preventDefault();
+      event.stopPropagation();
+
+      if (event.target.tagName.toLowerCase() !== 'input') {
+        this._onItemClick();
```
Comment (dkalinin) [inline app/view/component/file-list-item/index.js]: This is not needed when we handle click on name only.

**Pull request merged**